holoviews/core/data/dictionary.py

```diff
@@ -84,32 +85,39 @@ def init(cls, eltype, data, kdims, vdims):
 
         if not isinstance(data, cls.types):
             raise ValueError("DictInterface interface couldn't convert data.""")
-        elif isinstance(data, dict):
-            unpacked = []
-            for d, vals in data.items():
-                if isinstance(d, tuple):
-                    vals = np.asarray(vals)
-                    if vals.shape == (0,):
-                        for sd in d:
-                            unpacked.append((sd, np.array([], dtype=vals.dtype)))
-                    elif not vals.ndim == 2 and vals.shape[1] == len(d):
-                        raise ValueError("Values for %s dimensions did not have "
-                                         "the expected shape.")
-                    else:
-                        for i, sd in enumerate(d):
-                            unpacked.append((sd, vals[:, i]))
+
+        unpacked = []
```

Comment:
Note that I simply unindented this section when I thought I'd have to complicate it further, there are no changes to ``DictInterface.init`` that need reviewing.